client/server/pages/index.js

```diff
@@ -693,6 +694,30 @@ function wpcomPages( app ) {
 		}
 	} );
 
+	app.get( '/:locale/plans', function ( req, res ) {
```

Comment:
`:locale` param is currently matching anything, e.g. `/anything/plans`.

Let's use `import { getLanguageRouteParam } from '@automattic/i18n-utils';` here instead so that we make sure only locale slugs are matched for the route.